src/anbox/qemu/adb_message_processor.cpp

```diff
@@ -50,7 +54,11 @@ AdbMessageProcessor::AdbMessageProcessor(
 
 AdbMessageProcessor::~AdbMessageProcessor() {
   state_ = closed_by_host;
+
+  host_notify_timer_.cancel();
   host_connector_.reset();
+
+  // Unlock our lock to bring down any waiting instance
   active_instance.unlock();
```

Comment:
Please use a std::unique_lock<std::mutex> instead of manually unlocking the mutex. You can construct the lock with std::unique_lock<std::mutex> lock_{active_instance, std::defer_lock_t{}};

Minor niggle: active_instance -> active_instance_